### PR TITLE
Allow the vote plugin's position to be configured

### PR DESCRIPTION
**Pull request by mbabker**
#### Summary of Changes

Presently the voting plugin only triggers for the `onContentBeforeDisplay` event.  This PR adds the `onContentAfterDisplay` event as well and makes the position configurable.
#### Testing Instructions

Apply this patch from a checkout of the `3.7.x` branch and ensure the new configuration works correctly.

### CODE REVIEW

File: administrator/language/en-GB/en-GB.plg_content_vote.ini [hunk 1]
```diff
@@ -4,10 +4,14 @@
 ; Note : All ini files need to be saved as UTF-8
 
 PLG_CONTENT_VOTE="Content - Vote"
+PLG_VOTE_BOTTOM="Bottom"
 PLG_VOTE_LABEL="Please Rate"
+PLG_VOTE_POSITION_DESC="Set where the article's voting data is displayed"
```
Comment (brianteeman) [inline administrator/language/en-GB/en-GB.plg_content_vote.ini]: I'm not sure data is the right word. Doesn't seem right to me.

File: plugins/content/vote/vote.php [hunk 2]
```diff
@@ -33,7 +33,30 @@ class PlgContentVote extends JPlugin
...
+	 */
+	public function __construct(&$subject, $config)
+	{
+		parent::__construct($subject, $config);
+
+		$this->votingPosition = $this->params->get('position', 'top');
+	}
+
+	/**
+	 * Displays the voting area if in an article if displayed at the top of the article
```
Comment (brianteeman) [inline plugins/content/vote/vote.php]: I know this is a copy paste error but let's fix the if to of

File: plugins/content/vote/vote.php [hunk 3]
```diff
@@ -44,7 +67,51 @@ class PlgContentVote extends JPlugin
...
+		if ($this->votingPosition != 'top')
+		{
+			return '';
+		}
+
+		return $this->displayVotingData($context, $row, $params, $page);
+	}
+
+	/**
+	 * Displays the voting area if in an article if displayed at the bottom of the article
```
Comment (brianteeman) [inline plugins/content/vote/vote.php]: Is displayed not if displayed

File: administrator/language/en-GB/en-GB.plg_content_vote.ini (hunk 1 above)
Comment (mbabker) [inline administrator/language/en-GB/en-GB.plg_content_vote.ini]: It doesn't feel right to me either but I don't know what else to call it other than a section.

Comment (brianteeman) [inline administrator/language/en-GB/en-GB.plg_content_vote.ini]: How about  just simply

"Set where the voting is displayed."

**Comment by brianteeman** [comment]
Was there a specific reason this is against 3.7 and not staging? From experience it makes it harder for people to test<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/11375">issues.joomla.org/joomla-cms/11375</a>.</sub>

**Comment by mbabker** [comment]
Because the layout PR I did was merged to 3.7 so it wouldn't be very efficient to push this against staging then have to rebase it for 3.7 after it was tested against a completely different code structure.

**Comment by brianteeman** [comment]
Ok thanks. Makes sense.

**Comment by hardiktailored** [comment]
I have tested this item :white_check_mark: successfully on a4876321221f9187bbe9a563cd9d9b6542441e45<br /><br />Enabled votes for articles and displayed reviews, added ratings, showed on top and bottom position from plugin option, etc.
Tested both the events(onContentBeforeDisplay and onContentAfterDisplay) using echo out in code and everything works as expected.
Tested on 3.7.x(https://github.com/joomla/joomla-cms/archive/3.7.x.zip)<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/11375">issues.joomla.org/joomla-cms/11375</a>.</sub>

**Comment by 1apweb** [comment]
I have tested this item :white_check_mark: successfully on a4876321221f9187bbe9a563cd9d9b6542441e45<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/11375">issues.joomla.org/joomla-cms/11375</a>.</sub>

**Comment by bertmert** [comment]
I have tested this item :white_check_mark: successfully on a4876321221f9187bbe9a563cd9d9b6542441e45<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/11375">issues.joomla.org/joomla-cms/11375</a>.</sub>

**Comment by brianteeman** [comment]
@mbabker can you look at the travis issues please and then this can be merged<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/11375">issues.joomla.org/joomla-cms/11375</a>.</sub>

**Comment by mbabker** [comment]
The Travis issues exist in the 3.7.x branch and aren't related to this pull request.  https://travis-ci.org/joomla/joomla-cms/builds/148585025 is the last Travis build for the branch.

**Comment by brianteeman** [comment]
Thank @mbabker 

RTC <hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/11375">issues.joomla.org/joomla-cms/11375</a>.</sub>

**Comment by JoshuaLewis** [comment]
Still planned for J 3.7?<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/11375">issues.joomla.org/joomla-cms/11375</a>.</sub>

**Comment by brianteeman** [comment]
@JoshuaLewis yes<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/11375">issues.joomla.org/joomla-cms/11375</a>.</sub>

**Comment by wilsonge** [comment]
Merged with https://github.com/joomla/joomla-cms/commit/a6f5e84f123a807b7794faa8ee60c2118decc3cd

**Pull request closed without merging**